### PR TITLE
Consume cluster-level document-store configuration to consume MongoDB credentials

### PR DESCRIPTION
**Pull request by Vaibhav090420**
## Description
Please include a summary of the change, motivation and context.

### Testing
Please describe the tests that you ran to verify your changes. Please summarize what did you test and what needs to be tested e.g. deployed and tested helm chart locally. 

### Checklist:
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### Documentation
Make sure that you have documented corresponding changes in this repository or [hypertrace docs repo](https://github.com/hypertrace/hypertrace-docs-website) if required.

### CODE REVIEW

**Review by suresh-prakash — COMMENTED** [review]
Changes to mount the credentials seem to be missing in `deployment.yaml`.

File: partitioner-config-service-impl/src/main/java/org/hypertrace/partitioner/config/service/PartitionerConfigServiceFactory.java [hunk 1]
```diff
@@ -4,10 +4,18 @@
 import com.google.inject.Injector;
 import com.typesafe.config.Config;
 import io.grpc.BindableService;
+import org.hypertrace.core.serviceframework.spi.PlatformServiceLifecycle;
 
 public class PartitionerConfigServiceFactory {
+
+  @Deprecated
   public static BindableService build(Config config) {
```
[the remark this answers is not in the document]
Comment (Vaibhav090420) [reply partitioner-config-service-impl/src/main/java/org/hypertrace/partitioner/config/service/PartitionerConfigServiceFactory.java:12]: currently traceable-config-service is using this method, will remove once I change it there

File: partitioner-config-service-impl/src/main/java/org/hypertrace/partitioner/config/service/PartitionerConfigServiceModule.java [hunk 2]
```diff
@@ -16,24 +17,34 @@ public class PartitionerConfigServiceModule extends AbstractModule {
   public static final String PARTITIONER_CONFIG_SERVICE = "partitioner.config.service";
 
   private final Config config;
+  private PlatformServiceLifecycle lifecycle;
 
+  @Deprecated
   public PartitionerConfigServiceModule(Config config) {
```
[the remark this answers is not in the document]
Comment (Vaibhav090420) [reply partitioner-config-service-impl/src/main/java/org/hypertrace/partitioner/config/service/PartitionerConfigServiceModule.java:23]: currently used in traceable-config-service, will remove once changes are made there

File: config-service/src/integrationTest/resources/configs/config-service/application.conf [hunk 3]
```diff
@@ -6,7 +6,7 @@ service.admin.port = 50102
 
 generic.config.service {
   document.store {
-      appName = hypertrace-config-service-local
+      appName = config-service-config-test
```
Comment (suresh-prakash) [inline config-service/src/integrationTest/resources/configs/config-service/application.conf:9]:
```suggestion
      appName = config-service-test
```

File: config-service/src/main/resources/configs/common/application.conf [hunk 4]
```diff
@@ -5,7 +5,7 @@ service.admin.port = 50102
 
 generic.config.service {
   document.store {
-      appName = hypertrace-config-service-local
+      appName = config-service-config-local
```
Comment (suresh-prakash) [inline config-service/src/main/resources/configs/common/application.conf:8]:
```suggestion
      appName = config-service-local
```

**Review by suresh-prakash — COMMENTED** [review]
Wrapping `aggregate()` / `search()` calls under `try-with-resources` block seems to be missing. Apologies for missing this out in the previous iteration.

File: helm/templates/config-service-config.yaml [hunk 5]
```diff
@@ -11,15 +11,29 @@ data:
 
     generic.config.service {
       document.store {
-        {{- $dst := .Values.configServiceConfig.dataStoreType }}
+        {{- if .Values.database }}
+        {{- $dst := .Values.database.type }}
         dataStoreType = {{ $dst }}
+        appName = {{ .Values.configServiceConfig.name }}
```
Comment (suresh-prakash) [inline helm/templates/config-service-config.yaml:17]: This should be the service name and not the config name. Could you please introduce a separate section called `service` under value.yaml and add the name there to reference it?

**Comment by Vaibhav090420** [comment]
regarding this [https://github.com/hypertrace/config-service/pull/243#pullrequestreview-2374555340](url), all the 
`aggregate() / search()` are in `try-with-resource` block already

**Review by suresh-prakash — APPROVED** [review]
Looks good to me.

**Comment by suresh-prakash** [comment]
Please verify as the wiki suggests once the changes are live.

**Review by avinashkolluru — APPROVED** [review]

**Comment by mihirgt** [comment]
@Vaibhav090420 This PR is breaking traceable-config-service
https://github.com/Traceableai/config-service/actions/runs/11456871397/job/31875941718?pr=2261

**Pull request merged**